### PR TITLE
master-qa-60044

### PR DESCRIPTION
**Pull request by gaborlovasdotcom**
https://issues.liferay.com/browse/LRQA-60044

### CODE REVIEW

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
ci:test:bundle

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:sf

**Comment by gaborlovasdotcom** [comment]
ci:test:saml:nocompile

**Comment by gaborlovasdotcom** [comment]
CC @xbrianlee, could you check my current build-test.xml modification?
I would like to make it more reusable for anybody, who would like to deploy apps from different folders.
Many thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 1]
```diff
@@ -1092,6 +1092,55 @@ definition {
 		PortletEntry.publish();
 	}
 
+	@description = "This is a use case for LPS-110450."
+	@priority = "5"
+	test VerifyErrorMessageInSaas {
+		property osgi.app.includes = "saml-saas";
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:1098]: I thought `saml-saas` would be distributed somehow so I suggested building it as an LPKG (related: https://liferay.slack.com/archives/GLBT9FL23/p1596220342094500).`osgi.modules.includes` is the way to go in this case. I just tested it and it deploys the module successfully.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 2]
```diff
@@ -1092,6 +1092,55 @@ definition {
 		PortletEntry.publish();
 	}
 
+	@description = "This is a use case for LPS-110450."
+	@priority = "5"
+	test VerifyErrorMessageInSaas {
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:1097]: We should think about splitting out the SaaS test cases into a separate testcase file

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 3]
```diff
@@ -1092,6 +1092,55 @@ definition {
...
+		property osgi.app.includes = "saml-saas";
+		property test.name.skip.portal.instance = "SAML#ErrorMessageValidation";
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:1099]: This property needs to match the test name

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase [hunk 4]
```diff
@@ -1092,6 +1092,55 @@ definition {
...
+
+		ApplicationsMenu.gotoPortlet(
+			category = "System",
+			panel = "Control Panel",
+			portlet = "Virtual Instances");
+
+		PortalInstances.addCP(
+			mailDomain = "www.able.com",
+			virtualHost = "www.able.com",
+			webId = "www.able.com");
```
Comment (xbrianlee) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:1123]: Is it required to create a new portal instance? Most system settings affect the default portal instance.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase (hunk 1 above)
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase]: Thanks, I will modify this part.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase (hunk 2 above)
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:1097]: As we will write more Saas specific test case we will separate them.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase (hunk 3 above)
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase]: I update it.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase (hunk 4 above)
Comment (gaborlovasdotcom) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase:1123]: In this setup yes, we would like to mimic a real scenario whit a virtual instance, but I test how it works without it as well.

**Pull request closed without merging**